### ISSUE TITLE
Assert in Transforms.IntroduceUsingDeclarations.Run - using left tree navigation

### ISSUE THREAD

**Issue by greenozon** [icsharpcode/ILSpy#1254]
ILSpy version 4.0.0.4305-beta1

Note: it was already fixed in some older versions... but is back to life again!

```
ICSharpCode.ILSpy.AssertionFailedException: Should not insert using declaration for namespace that is missing from the superset: 
   at ICSharpCode.ILSpy.ILSpyTraceListener.Fail(String message, String detailMessage)
   at ICSharpCode.ILSpy.ILSpyTraceListener.Fail(String message)
   at System.Diagnostics.TraceInternal.Fail(String message)
   at System.Diagnostics.Debug.Assert(Boolean condition, String message)
   at ICSharpCode.Decompiler.CSharp.Transforms.IntroduceUsingDeclarations.Run(AstNode rootNode, TransformContext context)
   at ICSharpCode.Decompiler.CSharp.CSharpDecompiler.RunTransforms(AstNode rootNode, DecompileRun decompileRun, ITypeResolveContext decompilationContext)
   at ICSharpCode.Decompiler.CSharp.CSharpDecompiler.Decompile(IList`1 definitions)
   at ICSharpCode.Decompiler.CSharp.CSharpDecompiler.Decompile(EntityHandle[] definitions)
   at ICSharpCode.ILSpy.CSharpLanguage.DecompileMethod(IMethod method, ITextOutput output, DecompilationOptions options)
   at ICSharpCode.ILSpy.TreeNodes.MethodTreeNode.Decompile(Language language, ITextOutput output, DecompilationOptions options)
   at ICSharpCode.ILSpy.TextView.DecompilerTextView.DecompileNodes(DecompilationContext context, ITextOutput textOutput)
   at ICSharpCode.ILSpy.TextView.DecompilerTextView.<>c__DisplayClass34_0.<DecompileAsync>b__0()

```

target:  https://www.sendspace.com/file/28idne

err20.dll
almost all methods are asserting in it.

**Comment by siegfriedpammer** [comment]
I cannot reproduce this. Not using the current master branch and not on the pull request branch I created earlier.

Please be careful which version you use for testing. `4.0.0.4305-beta1` looks like a pull request build you got from AppVeyor.

**Comment by greenozon** [comment]
You are right,
ILSpy version 4.0.0.4285-beta1  (latest released bits here) does not have this  issue

but I"m wondering, how (where) could I get the latest bits and test things
isn't it AppVeyor location?

the message I'd like to b-cast is that latest bits most probably has some regression described above.

or there is another location of latest bits I'm not aware about?

**Comment by siegfriedpammer** [comment]
AppVeyor is the correct location. However there is no separate build for master and for pull requests. They all end up in the same list. If you take a look at https://ci.appveyor.com/project/icsharpcode/ilspy/history you can see that build 4305 is a PR build. Its name ends in '-prXXXX'. I just wanted to ask you to test the latest build from master branch, which is 4302.

Can you reproduce it with build 4302?

**Comment by greenozon** [comment]
Thanks for answer!
so AppVeyor is fine, I've taken from there build 4302 - http://prntscr.com/kh3e3i
and I"m able to reproduce the issue 100%

run - http://prntscr.com/kh3f88

oh..... looks like **the issue is more tricky!**  
save full projects works 100% without any assert at all!
but, the issue **is present** when you click on 1st API in 1st namespace/class by mouse! (also click on other methods - would get the same assert..)

http://prntscr.com/kh3fxb

this is  pretty much strange...  
generally speaking I'm confused that the user might get different asserts - one on save, others on navigating the code..
I'd like to understand why is that!

**Comment by greenozon** [comment]
To make the picture fully clean - clicking issue not present on latest released bits here - ILSpy version 4.0.0.4285-beta1
thus I make a conclusion that this is regression
Idea: would be nice to catch it by some tests... and not regress in the future

**Comment by siegfriedpammer** [comment]
> To make the picture fully clean - clicking issue not present on latest released bits here - ILSpy version 4.0.0.4285-beta1

I am sorry to disappoint you, but the released bits have this bug as well. It's just that release builds do not contain any assertions by definition. ;-)

So this is no regression. It cannot be, because the code never handled the special case we are running into.

**Comment by greenozon** [comment]
Oh, you right!
I forget that release bits are of "release type" meaning no debug asserts at all...

so lets keep this issue opened as a brand new? :)
